flow360/component/simulation/validation/validation_context.py

```diff
@@ -0,0 +1,119 @@
+"""
+Module for validation context handling in the simulation component of Flow360.
+
+This module provides context management for validation levels and specialized field
+definitions for conditional validation scenarios.
+
+Features
+--------
+- This module allows for defining conditionally (context-based) required fields,
+  such as fields required only for specific scenarios like surface mesh or volume mesh.
+- It supports running validation only for specific scenarios (surface mesh, volume mesh, or case),
+  allowing for targeted validation flows based on the current context.
+- This module does NOT ignore validation errors; instead, it enriches errors with context
+  information, enabling downstream processes to filter and interpret errors based on scenario-specific requirements.
+"""
+
+import contextvars
+from typing import Optional
+
+from pydantic import Field
+
+SURFACE_MESH = "SurfaceMesh"
+VOLUME_MESH = "VolumeMesh"
+CASE = "Case"
+# when running validation with ALL, it will report errors happing in all scenarios in one validation pass
+ALL = "All"
+
+_validation_level_ctx = contextvars.ContextVar("validation_level", default=None)
+
+
+class ValidationLevelContext:
+    """
+    Context manager for setting the validation level.
+
+    Allows setting a specific validation level within a context, which influences
+    the conditional validation of fields based on the defined levels.
+    """
+
+    def __init__(self, level: str):
+        if level not in {None, SURFACE_MESH, VOLUME_MESH, CASE, ALL}:
+            raise ValueError(f"Invalid validation level: {level}")
+        self.level = level
+        self.token = None
+
+    def __enter__(self):
+        self.token = _validation_level_ctx.set(self.level)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        _validation_level_ctx.reset(self.token)
+
+
+def get_validation_level():
+    """
+    Retrieves the current validation level from the context.
+
+    Returns:
+        The current validation level, which can influence field validation behavior.
+    """
+    return _validation_level_ctx.get()
+
+
+# pylint: disable=invalid-name
+def ConditionalField(
```

Comment:
Recommend renaming:
```
ContextField is alias of ConditionalField(required=False, context=XXX)
ConditionalField is alias of ConditionalField(required=True, context=XXX)
```
Main gripe is that when `required=False` there is nothing "Conditional" about the field, it just try to provide some context.

Reply:
fixed in: https://github.com/flexcompute/Flow360/commit/93803d6b5585f66f88708ef31130d01c0267af98